### ISSUE TITLE
Bad coloring of CRTP inheritance with templated child class

### ISSUE THREAD

**Issue by kylrth** [jeff-hykin/cpp-textmate-grammar#314]
## To reproduce

Type this in a C++ header file (e.g. `sample.hpp`) in VS Code:

```cpp
template <typename T, class Child>
class BaseClass {};

template <typename T>
class ChildClass : public BaseClass<T, ChildClass<T>> {};  // "BaseClass" colored incorrectly

class OtherChild : public BaseClass<int, OtherChild> {};  // "BaseClass" colored correctly
```

## Expected

`BaseClass` should be colored with the template color in the definition of `ChildClass`.

## Actual

`BaseClass` is colored the same color as variable names and parentheses.

## Screenshot

![badHighlighting](https://user-images.githubusercontent.com/5044802/61385029-590b5400-a877-11e9-8bb9-861c46f1d7f4.png)

## Configuration

Extension version: 1.12.20
C/C++ extension version: 0.24.0
VS Code version: 1.37.0-insider (user setup)
Commit: 8e0f348413f4f616c23a88ae30030efa85811973
Date: 2019-07-16T05:25:39.655Z
OS: Windows_NT x64 6.3.9600 (Windows 8)

**Comment by matter123** [comment]
@jeff-hykin The correct solution is to replace the entire `template_call` expression with`(?<template>(?<!<)<(?!<)(?:[^<>]++(?:\g<template>)?)++>)`. I did not do that in #308 as subexp calls are still untested. Additionally that creates a new capture group that needs to be accounted for.

The current fix in #308 allows for 0 or 1 levels of template nesting.

**Comment by matter123** [comment]
@kylrth This will be fixed when #308 is merged in, presumably sometime after the 23rd. If you need the fix for this or #307 now. You can replace the file located at `<vscode-ext-path>/jeff-hykin.better-cpp-syntax-1.12.20/syntaxes/cpp.tmLanguage.json` with https://raw.githubusercontent.com/jeff-hykin/cpp-textmate-grammar/Fix/%23307/syntaxes/cpp.tmLanguage.json

**Comment by kylrth** [comment]
Great! I can be patient with my code coloring until the 23rd. :)
